nucleus/admin/monitor/src/main/java/fish/payara/admin/monitor/cli/ResetMonitoringLevels.java

```diff
@@ -0,0 +1,136 @@
+import com.sun.enterprise.config.serverbeans.MonitoringService;
+import com.sun.enterprise.util.ColumnFormatter;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.inject.Inject;
+import org.glassfish.api.ActionReport;
+import org.glassfish.api.Param;
+import org.glassfish.api.admin.AdminCommand;
+import org.glassfish.api.admin.AdminCommandContext;
+import org.glassfish.api.admin.ExecuteOn;
+import org.glassfish.api.admin.RuntimeType;
+import org.glassfish.config.support.CommandTarget;
+import org.glassfish.config.support.TargetType;
+import org.glassfish.hk2.api.PerLookup;
+import org.glassfish.internal.api.Target;
+import org.jvnet.hk2.annotations.Service;
+import org.jvnet.hk2.config.ConfigSupport;
+import org.jvnet.hk2.config.TransactionFailure;
+
+/**
+ *
+ * @author Alan Roth
+ */
+@Service(name = "reset-monitoring-levels")
+@PerLookup
+@ExecuteOn({RuntimeType.DAS, RuntimeType.INSTANCE})
+@TargetType({CommandTarget.DAS, CommandTarget.STANDALONE_INSTANCE, CommandTarget.CLUSTER, CommandTarget.CONFIG})
+public class ResetMonitoringLevels implements AdminCommand {
+
+    @Param(name = "target", optional = true, defaultValue = "server-config")
+    String target;
+
+    @Param(name = "verbose", optional = true, shortName = "v")
+    private boolean verbose;
+
+    @Inject
+    private Target targetUtil;
+
+    @Inject
+    private Logger logger;
+
+    private MonitoringService monitoringService;
+
+    @Override
+    public void execute(AdminCommandContext context) {
+        ActionReport actionReport = context.getActionReport();
+
+        List<String> validModuleList = Arrays.asList(Constants.validModuleNames);
+        Config config = targetUtil.getConfig(target);
+        if (config != null) {
+            monitoringService = config.getMonitoringService();
+        } else {
+            actionReport.setMessage("Cound not find target: " + target);
+            actionReport.setActionExitCode(ActionReport.ExitCode.FAILURE);
+            return;
+        }
+
+        Map<String, String> enabledModules = getPreviouslyEnabledModules(validModuleList);
+
+        //Setting all modules to "OFF"
+        for (String module : validModuleList) {
+            try {
+                ConfigSupport.apply((final MonitoringService monitoringServiceProxy) -> {
+                    monitoringServiceProxy.setMonitoringLevel(module, "OFF");
+                    return monitoringServiceProxy;
+                }, monitoringService);
+            } catch (TransactionFailure ex) {
+                logger.log(Level.WARNING, "Failed to execute the command set-monitoring-level, while setting modules to 'OFF': {0}", ex.getCause().getMessage());
+                actionReport.setMessage(ex.getCause().getMessage());
+                actionReport.setActionExitCode(ActionReport.ExitCode.FAILURE);
```

Comment:
If you're going to carry on execution, might it be better to set the exit code to warning so that the output is "command completed with warnings" (or w/e the exact string is)?

Reply:
Good idea